### PR TITLE
feat: Updated SS and microservices deployment docs

### PR DESCRIPTION
**Pull request by Jayclifford345**
This PR updates our helm documentation inline with our 3.0 deployment for:
- Simple Scalable
- Microservices

**Checklist**
- [x] Reviewed the [`CONTRIBUTING.md`](https://github.com/grafana/loki/blob/main/CONTRIBUTING.md) guide (**required**)
- [x] Documentation added
- [ ] Tests updated
- [ ] Title matches the required conventional commits format, see [here](https://www.conventionalcommits.org/en/v1.0.0/)
  - **Note** that Promtail is considered to be feature complete, and future development for logs collection will be in [Grafana Alloy](https://github.com/grafana/alloy). As such, `feat` PRs are unlikely to be accepted unless a case can be made for the feature actually being a bug fix to existing behavior.
- [ ] Changes that require user attention or interaction to upgrade are documented in `docs/sources/setup/upgrade/_index.md`
- [ ] For Helm chart changes bump the Helm chart version in `production/helm/loki/Chart.yaml` and update `production/helm/loki/CHANGELOG.md` and `production/helm/loki/README.md`. [Example PR](https://github.com/grafana/loki/commit/d10549e3ece02120974929894ee333d07755d213)
- [ ] If the change is deprecating or removing a configuration option, update the `deprecated-config.yaml` and `deleted-config.yaml` files respectively in the `tools/deprecated-config-checker` directory. [Example PR](https://github.com/grafana/loki/pull/10840/commits/0d4416a4b03739583349934b96f272fb4f685d15)

### CODE REVIEW

File: docs/sources/setup/install/helm/install-microservices/_index.md [hunk 1]
```diff
@@ -0,0 +1,223 @@
+---
+title: Install the microservice Helm chart 
+menuTitle: Install microservice Loki
+description: Installing Loki in microservice (distributed) mode using the Helm chart.
+aliases:
+  - ../../../installation/helm/microservice/
+  - ../../../installation/helm/install-microservice/
```
Comment (JStickler) [inline docs/sources/setup/install/helm/install-microservices/_index.md:7]: Since this page has never existed before, I doesn't need aliases.  Those are for creating redirects when an existing page has moved or been renamed.

File: docs/sources/setup/install/helm/install-microservices/_index.md [hunk 2]
```diff
@@ -0,0 +1,223 @@
...
+weight: 300
+keywords: 
+---
+
+# Install the microservice Helm chart
+<!-- vale Grafana.Quotes = NO -->
+<!-- vale Grafana.Quotes = YES -->
```
Comment (JStickler) [inline docs/sources/setup/install/helm/install-microservices/_index.md:14]: These are only necessary if you're skipping a Vale rule.  Since there's nothing between the comments, you're not skipping a rule and can remove these.

File: docs/sources/setup/install/helm/install-microservices/_index.md [hunk 3]
```diff
@@ -0,0 +1,223 @@
...
+
+This Helm Chart deploys Grafana Loki on Kubernetes.
+
+This chart configures Loki to run Loki in [microservice / distributed mode]({{< relref "../../../../get-started/deployment-modes#microservices-mode" >}}), highly available architecture designed to work with AWS S3 object storage.
```
Comment (JStickler) [inline docs/sources/setup/install/helm/install-microservices/_index.md:18]: I don't think microservices mode is specific to any particular/specific object storage. Maybe reword that this chart is configured for AWS S3 by default, but can easily be edited for GCS and Azure?

File: docs/sources/setup/install/helm/install-microservices/_index.md [hunk 4]
```diff
@@ -0,0 +1,223 @@
...
+
+# Install the microservice Helm chart
+<!-- vale Grafana.Quotes = NO -->
+<!-- vale Grafana.Quotes = YES -->
+
+This Helm Chart deploys Grafana Loki on Kubernetes.
+
+This chart configures Loki to run Loki in [microservice / distributed mode]({{< relref "../../../../get-started/deployment-modes#microservices-mode" >}}), highly available architecture designed to work with AWS S3 object storage. 
+
+The default Helm chart deploys the following components:
```
Comment (JStickler) [inline docs/sources/setup/install/helm/install-microservices/_index.md:20]: Whenever you have a list of items, unless there's a really good reason not to, you should alphabetize the list.

File: docs/sources/setup/install/helm/install-microservices/_index.md [hunk 5]
```diff
@@ -0,0 +1,223 @@
...
+- **Ingester component** (3 replicas): Handles ingestion of data.
+- **Querier component** (3 replicas, maxUnavailable: 2): Processes queries. Up to 2 replicas can be unavailable during updates.
+- **QueryFrontend component** (2 replicas, maxUnavailable: 1): Manages frontend queries. Up to 1 replica can be unavailable during updates.
+- **QueryScheduler component** (2 replicas): Schedules queries.
+- **Distributor component** (3 replicas, maxUnavailable: 2): Distributes incoming requests. Up to 2 replicas can be unavailable during updates.
+- **Compactor component** (1 replica): Compacts and processes stored data.
+- **IndexGateway component** (2 replicas, maxUnavailable: 1): Handles indexing. Up to 1 replica can be unavailable during updates.
+
+
+<!--TODO - Update when meta-monitoring chart releases-->
```
Comment (JStickler) [inline docs/sources/setup/install/helm/install-microservices/_index.md:30]: FYI - It has been released, just not publicized yet.

https://github.com/grafana/meta-monitoring-chart/releases

File: docs/sources/setup/install/helm/install-microservices/_index.md [hunk 6]
```diff
@@ -0,0 +1,223 @@
...
+
+It is not recommended to run microservices mode with `filesystem` storage.
+
+**Prerequisites**
+
+- Helm 3 or above. See [Installing Helm](https://helm.sh/docs/intro/install/).
+- A running Kubernetes cluster.
+- (Optional) A Memcached deployment for better query performance. For information on configuring Memcached, refer to [caching section]({{< relref "../../../../operations/caching" >}}).
```
Comment (JStickler) [inline docs/sources/setup/install/helm/install-microservices/_index.md:38]:
```suggestion
- (Optional) A Memcached deployment for better query performance. For information on configuring Memcached, refer to the [caching section](https://grafana.com/docs/loki/<LOKI_VERSION>/operations/caching/).
```

File: docs/sources/setup/install/helm/install-microservices/_index.md [hunk 7]
```diff
@@ -0,0 +1,223 @@
...
+    - To upgrade:
+       ```bash
+       helm upgrade --values values.yaml loki grafana/loki
+       ```
+       
+
+1. Verify that Loki is running:
+    ```bash
+    kubectl get pods -n loki
+    ```
```
Comment (JStickler) [inline docs/sources/setup/install/helm/install-microservices/_index.md:147]: I think it's helpful to add example output from this command so that users know what pods they should expect to see.

File: docs/sources/setup/install/helm/install-microservices/testing-deployment-with-kind.md [hunk 8]
```diff
@@ -0,0 +1,176 @@
+---
+title: Test the microservice Helm chart with Kind
+menuTitle: Test the microservice Helm chart with Kind
+description: Installing Loki in microservice (distributed) mode using the Helm chart.
+aliases:
+  - ../../../installation/helm/test-microservice/
+  - ../../../installation/helm/test-install-microservice/
```
Comment (JStickler) [inline docs/sources/setup/install/helm/install-microservices/testing-deployment-with-kind.md:7]: Same comment as above, since this is a new page, it should not have aliases.

File: docs/sources/setup/install/helm/install-microservices/testing-deployment-with-kind.md [hunk 9]
```diff
@@ -0,0 +1,176 @@
...
+weight: 300
+keywords: 
+---
+
+
+# Test the microservice Helm chart with Kind
+<!-- vale Grafana.Quotes = NO -->
+<!-- vale Grafana.Quotes = YES -->
```
Comment (JStickler) [inline docs/sources/setup/install/helm/install-microservices/testing-deployment-with-kind.md:15]: Same comment as above, these are unnecessary.

File: docs/sources/setup/install/helm/install-microservices/testing-deployment-with-kind.md [hunk 10]
```diff
@@ -0,0 +1,176 @@
...
+
+This guide walks you through testing the Grafana Loki Helm chart in microservice mode using [Kind](https://kind.sigs.k8s.io/), a tool for running local Kubernetes clusters using Docker container nodes.
+
+## Prerequisites
+
+To run through this tutorial, you need the following tools installed:
```
Comment (JStickler) [inline docs/sources/setup/install/helm/install-microservices/testing-deployment-with-kind.md:21]: Not exactly sure if we need this topic, since we don't have a "testing" topic for the other installation methods.  I would assume that customers would test in a test environment before installing on a production environment.  Have you seen different behavior?

File: docs/sources/setup/install/helm/install-scalable/_index.md [hunk 11]
```diff
@@ -37,7 +37,7 @@ It is not recommended to run scalable mode with `filesystem` storage.
 - (Optional) A Memcached deployment for better query performance. For information on configuring Memcached, refer to [caching section]({{< relref "../../../../operations/caching" >}}).
 
 
-**To deploy Loki in simple scalable mode:**
+**To deploy Loki in microservices mode:**
```
Comment (JStickler) [inline docs/sources/setup/install/helm/install-scalable/_index.md:40]: Is this a copy/paste error?  You have a separate topic for microservices mode.

File: docs/sources/setup/install/helm/install-scalable/_index.md [hunk 12]
```diff
@@ -46,40 +46,83 @@ It is not recommended to run scalable mode with `filesystem` storage.
    helm repo add grafana https://grafana.github.io/helm-charts
    ```
 
-1. Update the chart repository:
+2. Update the chart repository:
```
Comment (JStickler) [inline docs/sources/setup/install/helm/install-scalable/_index.md:49]: Unless this was numbered incorrectly in the output, best practice is to number all steps as 1.

https://grafana.com/docs/writers-toolkit/write/markdown-guide/#numbered-lists

File: docs/sources/setup/install/helm/install-scalable/_index.md [hunk 13]
```diff
@@ -46,40 +46,83 @@ It is not recommended to run scalable mode with `filesystem` storage.
...
+                period: 24h
+        ingester:
+          chunk_encoding: snappy
+        tracing:
+          enabled: true
+        querier:
+          # Default is 4, if you have enough memory and CPU you can increase, reduce if OOMing
+          max_concurrent: 4
+
+      #gateway:
```
Comment (JStickler) [inline docs/sources/setup/install/helm/install-scalable/_index.md:78]: Why include this in the example if it is commented out?

File: docs/sources/setup/install/helm/install-scalable/_index.md [hunk 14]
```diff
@@ -91,5 +134,77 @@ It is not recommended to run scalable mode with `filesystem` storage.
        helm upgrade --values values.yaml loki grafana/loki
        ```
 
+## AWS S3 Configuration
+
+To configure Loki to use AWS S3 as the object storage, you need to provide the following values in the `values.yaml` file:
```
Comment (JStickler) [inline docs/sources/setup/install/helm/install-scalable/_index.md:139]: Why not combine this with the example above?  Curious as to why it's called out separately.

**Review by JStickler — COMMENTED** [review]
[docs team]

File: docs/sources/setup/install/helm/install-microservices/_index.md (hunk 5 above)
Comment (Jayclifford345) [reply docs/sources/setup/install/helm/install-microservices/_index.md]: This is really cool!! I noticed we had a monitoring and alerting section in deployments. I have linked to that for now. I can create a new PR with updated instructions for that section of the docs with the repo instructions

File: docs/sources/setup/install/helm/install-microservices/testing-deployment-with-kind.md (hunk 10 above)
Comment (Jayclifford345) [reply docs/sources/setup/install/helm/install-microservices/testing-deployment-with-kind.md:17]: So I have changed the name slightly to help. Essentially I wanted to provide our community with an easy to step by step guide on how to practise deploying this locally just so they can understand how the helm deploys without needing to set up a full cloud k8's stack. It mostly helped me as I was testing the new deployment helms that the team created :). But happy to remove if its overkill as might also be confusing

File: docs/sources/setup/install/helm/install-scalable/_index.md (hunk 14 above)
Comment (Jayclifford345) [reply docs/sources/setup/install/helm/install-scalable/_index.md]: So I was going to ask you about this one. I currently have the first example using MinIO for test purposes and then provide the cloud storage configurations after. I have added the S3 one for now but currently testing Azure blog to also add next

**Comment by Jayclifford345** [comment]
Marking this as ready for review as we are getting quite a few questions in the community over helm deployment. Will create subsequent PR's with other further additions.

**Review by trevorwhitney — COMMENTED** [review]
this looks good, but I don't think we should include the instructions for deploying with kind. I think that will just lead to questions about kind we are not experts in. There's a lot of ways to spin up a local k8s cluster, I think it's reasonable to expect someone familiar with helm to bring their own k8s cluster.

**Comment by Jayclifford345** [comment]
@JStickler @trevorwhitney , that's the kind deployment instructions deleted. Let me know if there are any other parts you think might need some rework

**Review by trevorwhitney — APPROVED** [review]

File: docs/sources/setup/install/helm/install-microservices/testing-deployment-with-kind.md (hunk 10 above)
Comment (JStickler) [reply docs/sources/setup/install/helm/install-microservices/testing-deployment-with-kind.md]: LOL, just saw Trevor's comment and that you'd pulled it. That's OK too.
